Elements/src/Geometry/Polygon.cs

```diff
@@ -468,6 +516,11 @@ internal bool Contains3D(Vector3 point, bool unique = false)
             return intersects % 2 != 0;
         }
 
+        internal bool Contains3D(Polygon polygon)
+        {
+            return polygon.Vertices.All(v => this.Contains3D(v));
```

Comment:
It might not be enough to only check vertices in non-convex cases. Consider to check edge intersections, see image.

![afbeelding](https://user-images.githubusercontent.com/1096535/151129733-4e6d9380-78a9-4871-b667-f48868f5d670.png)

Reply:
Thanks very much Thomas! That's a great catch!